android/source/VideoLocker/src/org/edx/mobile/base/MainApplication.java

```diff
@@ -63,6 +64,9 @@ private void init() {
         if (Config.getInstance().getFacebookAppId() != null) {
             com.facebook.Settings.setApplicationId(Config.getInstance().getFacebookAppId());
         }
+
+        // try data repair if app version is updated
+        new Storage(this).doRepairData();
```

Comment:
Maybe give this a more specific name. Something like ``repairDownloadCompletionData``?